hr_timesheet_sheet/models/hr_timesheet_sheet.py

```diff
@@ -210,28 +336,49 @@ def _check_start_end_dates(self):
                     _('The start date cannot be later than the end date.'))
 
     @api.multi
-    def _get_overlapping_sheet_domain(self, user_id=False):
+    def _get_complete_name_components(self):
+        """ Hook for extensions """
+        self.ensure_one()
+        result = [self.employee_id.name_get()[0][1]]
+        if self.review_policy == 'department_manager':
+            result += [self.department_id.name_get()[0][1]]
+        elif self.review_policy == 'project_manager':
+            result += [self.project_id.name_get()[0][1]]
+        return result
+
+    @api.multi
+    def _get_overlapping_sheet_domain(self):
```

Comment:
@mreficent @astirpe to give you better insights: this method has another 4 lines of code specific to validation models.